### PR TITLE
test: Permanently disable net.ifnames on RHEL/CentOS/Fedora

### PR DESCRIPTION
**Pull request by martinpitt**
Calling grubby with --args isn't persistent as it doesn't update the config in /etc, so the next grub update removes those again. This happes in check-kdump (it modifies /etc/default/grub and calls grub-mkconfig), thus the next reboot would use systemd interface names again.

Replace the grubby call with setting the option in /etc/default/grub instead.

Drop the redundant grubby and grub-mkconfig call from fedora-*.setup, as virt-*-fedora already do that.

 * [x] image-refresh fedora-23-stock
 * [x] image-refresh fedora-24
 * [x] image-refresh fedora-25
 * [x] image-refresh fedora-26
 * [x] image-refresh centos-7
 * [x] image-refresh rhel-7
 * [x]  image-refresh openshift
 * [x] image-refresh ipa
 * [x] image-refresh selenium

### CODE REVIEW

**Comment by martinpitt** [comment]
So bumping the timeout wasn't it, it seems it genuinely fails to reboot, but only on the CI infra.

**Comment by martinpitt** [comment]
Turns out I actually *can* reproduce the reboot failure locally, sometimes. I ran a "virsh console" in parallel with the test, and it reboots just fine. I don't think there's anything wrong with the actual `reboot` command (even though it is unnecessarily complicated with the threefold sync and backgrounding), but that instead `wait_reboot()` itself isn't reliable, particularly this bit:
```py
       if not self.wait_execute(timeout_sec=wait_for_running_timeout, get_new_address=lambda: self._ip_from_mac(self.macaddr, timeout_sec=5)):
            raise Failure("system didn't reboot properly")
```

**Comment by martinpitt** [comment]
While the machine does reboot, there is no ssh@.service starting up, and setting up the network fails:
```
# systemctl status network.service|cat
● network.service - LSB: Bring up/down networking
   Loaded: loaded (/etc/rc.d/init.d/network; bad; vendor preset: disabled)
   Active: failed (Result: exit-code) since Tue 2017-06-13 09:19:46 EDT; 36s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 622 ExecStart=/etc/rc.d/init.d/network start (code=exited, status=1/FAILURE)

Jun 13 09:19:45 localhost.localdomain systemd[1]: Starting LSB: Bring up/down networking...
Jun 13 09:19:46 localhost.localdomain network[622]: Bringing up loopback interface:  [  OK  ]
Jun 13 09:19:46 localhost.localdomain network[622]: Bringing up interface eth0:  Error: Connection activation failed: No suitable device found for this connection.
Jun 13 09:19:46 localhost.localdomain network[622]: [FAILED]

2: ens5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:12:34:56 brd ff:ff:ff:ff:ff:ff
3: ens6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:01:00:00:01 brd ff:ff:ff:ff:ff:ff
4: virbr0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN qlen 1000
    link/ether 52:54:00:f8:fa:ca brd ff:ff:ff:ff:ff:ff
    inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0
       valid_lft forever preferred_lft forever
5: virbr0-nic: <BROADCAST,MULTICAST> mtu 1500 qdisc pfifo_fast master virbr0 state DOWN qlen 1000
    link/ether 52:54:00:f8:fa:ca brd ff:ff:ff:ff:ff:ff
```

So after reboot eth0 and eth1 are renamed ens5/6 and thus don't get an address. There is no "net.ifnames=0 biosdevname=0" in the kernel command line any more. The problem seems to be that calling `grubby --update-kernel=ALL --args="net.ifnames=0 biosdevname=0"` only seems to add the new parameters to the generated /boot/grub2/grub.cfg, but nowhere in the primary config files in /etc. So as soon as the grub config is being regenerated (which seems to be the case in rhel-7), it is gone. So either we can call grubby in a way to make this setting persistent, or we should perhaps `touch /etc/systemd/network/99-default.link` and regenerate the initrd, to be persistent and more platform independent.

**Comment by martinpitt** [comment]
Pushed a test commit that applies the setting properly in rhel-7.install. Of course we really want to do that in .install, this is just for verifying that it works on the infrastructure. It does work fine for me now locally.

**Comment by martinpitt** [comment]
Ah, hah! check-kdump itself does that:
```
           lines = self.machine.execute(command="cat /etc/default/grub", quiet=True).split("\n")
            lines = map(lambda line: self.rreplace(line, '"', ' crashkernel=256M"', 1) if line.startswith("GRUB_CMDLINE_LINUX") else line, lines)
            self.machine.write("/etc/default/grub", "\n".join(lines))
            self.machine.execute("grub2-mkconfig -o /boot/grub2/grub.cfg")
```
This explains why the persistent interface names come back after reboot, as that overrides the grubby changes. It's a miracle that the test actually didn't fail all the time since then..

**Comment by martinpitt** [comment]
check-kdump succeeded now. I re-pushed to fix this more properly with image rebuilds.

File: bots/images/scripts/virt-install-fedora [hunk 1]
```diff
@@ -64,16 +64,13 @@ chmod 600 /root/.ssh/authorized_keys
 mkdir -p /etc/sysconfig/network-scripts
 echo "$(cat $base/network-ifcfg-eth0)" > /etc/sysconfig/network-scripts/ifcfg-eth0
 echo "$(cat $base/network-ifcfg-eth1)" > /etc/sysconfig/network-scripts/ifcfg-eth1
-sed -i "s/GRUB_TIMEOUT.*/GRUB_TIMEOUT=0/" /etc/default/grub
```
Comment (dperpeet) [inline bots/images/scripts/virt-install-fedora]: What replaces this in `virt-install`?

Comment (martinpitt) [inline bots/images/scripts/virt-install-fedora]: Fixed, thanks!

**Comment by stefwalter** [comment]
@martinpitt Take off the WIP prefix in order to get bots interested in doing the tasks in the checklist.

**Comment by dperpeet** [comment]
> image-refresh fedora-23-stock

Why would we want to rebuild the stock image? Also I think fedora-24 doesn't need a rebuild for this.

**Comment by martinpitt** [comment]
> Why would we want to rebuild the stock image? Also I think fedora-24 doesn't need a rebuild for this.

Mostly because we usually want our images to coincide with the build scripts. fedora-23-stock.bootstrap and fedora-24.bootstrap call virt-builder-fedora. I don't mind skipping them if you say it's alright, I just thought to do that for completeness.

**Comment by cockpituous** [comment]
image-refresh in progress on verifymachine4.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-172121/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-172121/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-ctb1h.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-173434/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-84ht4.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-173535/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-173535/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-k9p63.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-174848/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-hjl9f.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-174848/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-nq3j8.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-174848/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-174848/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-183636/

**Comment by cockpituous** [comment]
image-refresh fedora-25 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-25-20170613-191010

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-84ht4.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-191111/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-191111/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-3zw9c.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-192020/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-3zw9c.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-200202/

**Comment by cockpituous** [comment]
image-refresh ipa done: https://github.com/cockpituous/cockpit/commits/image-refresh-ipa-20170613-202929

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-0xj8f.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-203131/

**Comment by cockpituous** [comment]
image-refresh fedora-23-stock done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-23-stock-20170613-205050

**Comment by dperpeet** [comment]
> Mostly because we usually want our images to coincide with the build scripts. fedora-23-stock.bootstrap and fedora-24.bootstrap call virt-builder-fedora. I don't mind skipping them if you say it's alright, I just thought to do that for completeness.

Like I said on irc, I think we should be able to safely skip the stock images. Yet it shouldn't be harmful to update it.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-1q07b.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-210303/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-210303/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-66zrn.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-211515/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-bgwxb.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-211515/

**Comment by martinpitt** [comment]
On the infra, openshift refreshes keep failing like this:
```
++ LC_ALL=C.UTF-8
++ python3 -c 'import json, sys; obj=json.load(sys.stdin); releases = [x.get('\''tag_name'\'', '\'''\'') for x in obj if not x.get('\''prerelease'\'')]; print(sorted (releases, reverse=True)[0])'
++ curl -s https://api.github.com/repos/openshift/origin/releases
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <listcomp>
AttributeError: 'str' object has no attribute 'get'
+ VERSION=
```
This doesn't happen locally - apparently on our infra the output of that curl is somehow different.

But on my laptop it fails after bootstrapping already with
```
[root@localhost ~]# [1497389035.23] EVENT: Domain 'openshift_f1.cockpit.lan' (ID 4) Shutdown Finished
image-create: Unable to reach machine 10.111.119.12 via ssh.
```

As this is unrelated and not blocking the network/kdump tests, would it be okay to investigate this separately?

**Comment by cockpituous** [comment]
image-refresh centos-7 done: https://github.com/cockpituous/cockpit/commits/image-refresh-centos-7-20170613-215252

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-bgwxb.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-215353/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-215353/

**Comment by martinpitt** [comment]
Fedora 26 often fails with
```
error: db5 error(5) from dbenv->open: Input/output error
error: cannot open Packages index using db5 - Input/output error (5)
error: cannot open Packages database in /var/lib/rpm
Error: Error: rpmdb open failed
```
I keep retrying.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-zz387.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-220606/

**Comment by cockpituous** [comment]
image-refresh selenium done: https://github.com/cockpituous/cockpit/commits/image-refresh-selenium-20170613-222525

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170613-225656/

**Comment by cockpituous** [comment]
image-refresh rhel-7 done: https://github.com/cockpituous/cockpit/commits/image-refresh-rhel-7-20170613-232020

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-wwd8c.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170614-060909/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-t8nth.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170614-061010/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-bgwxb.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170614-061010/

**Comment by cockpituous** [comment]
image-refresh fedora-25 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-25-20170614-065252

**Comment by martinpitt** [comment]
Indeed the rebuilt fedora 23 stock image is completely broken:
```
[[    3.505771] XFS (vda3): Internal error XFS_WANT_CORRUPTED_GOTO at line 1741 of file fs/xfs/libxfs/xfs_alloc.c.  Caller xfs_free_extent+0xb2/0x140 [xfs]
```
I'll try a rebuild, or just leave it out there (as @dperpeet suggested) if it continues to fail, it shouldn't block this PR and #6334.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-t8nth.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170614-085454/

**Comment by cockpituous** [comment]
image-refresh fedora-23-stock done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-23-stock-20170614-093030

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tests-hjl9f.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6923-20170614-093131/

**Comment by martinpitt** [comment]
That rebuilt fedora-23-stock image now works, unsure what caused the file system corruption on the previous one. Local build of fedora-26 works fine, I'm uploading that. A local openshift image rebuild also works (my previous woes were due to some libvirt regression on fedora-26, to be worked around with re-running `vm-prep` and `echo 2 | sudo tee /proc/sys/net/ipv6/conf/all/accept_ra`).

**Comment by martinpitt** [comment]
@dperpeet : The two failures on ubuntu-stable are due to the known d-bus "too many connections" issue, and this PR doesn't touch the ubuntu images anyway. The rest is green, so I'd appreciate if this could land soon. Then this unblocks #6334 and hopefully also #6360 .

**Pull request merged**